lib/engine/tile.rb

```diff
@@ -415,9 +423,9 @@ def separate_parts
         end
       end
 
-      @nodes = @paths.map(&:node).compact.uniq
-      @branches = @paths.map(&:branch).compact.uniq
-      @stops = @paths.map(&:stop).compact.uniq
+      @nodes = @paths.flat_map(&:nodes).compact.uniq
```

Comment:
i believe none of these should need compact, can you check?

Reply:
Yep, I'll check.

Reply:
#compact does appear to be superfluous here.